spec/rubocop/cop/style/map_compact_with_conditional_block_spec.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::MapCompactWithConditionalBlock, :config do
+  context 'With multiline block' do
+    it 'registers an offense and corrects to `select` with `if` condition' do
+      expect_offense(<<~RUBY)
+        foo.map do |item|
+            ^^^^^^^^^^^^^ Replace `map { ... }.compact` with `select`.
```

Comment:
OK, maybe the fix suggestion will be clearer if the block is shown.

```suggestion
            ^^^^^^^^^^^^^ Replace `map { ... }.compact` with `select { ... }`.
```

Reply:
I agree. Updated :)